src/server/req_runjob.c

```diff
@@ -502,36 +502,45 @@ req_runjob(struct batch_request *preq)
 		return;
 
 	} else if (jt == IS_ARRAY_Single) {
+		attribute sub_runcount = {0};
+		attribute sub_run_version = {0};
 
 		/* single subjob, if parent qeueud, it can be run */
 
-		if ((pjobsub = parent->ji_ajtrk->tkm_tbl[offset].trk_psubjob) != NULL)
+		if ((pjobsub = parent->ji_ajtrk->tkm_tbl[offset].trk_psubjob) != NULL) {
+			sub_runcount = pjobsub->ji_wattr[JOB_ATR_runcount];
+			sub_run_version = pjobsub->ji_wattr[JOB_ATR_run_version];
 			job_purge(pjobsub);
+		}
 
 		if ((pjobsub = create_subjob(parent, jid, &j)) == NULL) {
 			req_reject(j, 0, preq);
 			return;
 		}
+
+		if (sub_run_version.at_flags & ATR_VFLAG_SET) {
+			pjobsub->ji_wattr[(int)JOB_ATR_run_version].at_val.at_long = sub_run_version.at_val.at_long;
+			pjobsub->ji_wattr[(int)JOB_ATR_run_version].at_flags |= (ATR_VFLAG_SET | ATR_VFLAG_MODCACHE);
```

Comment:
Why don't you set ATR_VFLAG_MODIFY on this attribute?

Reply:
It doesn't matter as we will be doing full save later. Frankly I go this line form other places, where also it is missing.

Reply:
I feel it's better to add the MODIFY flag so there is no issue even if we move the full save logic in the future.

Reply:
Added the flag in https://github.com/PBSPro/pbspro/pull/1119/commits/982610ec6b04f71ab4769eca4907fa8c612eb884

Reply:
A similar change is needed on line 588 as well

Reply:
Addressed in https://github.com/PBSPro/pbspro/pull/1119/commits/ec73366673c4c1438ad7dfb4ae574888aa17c57b